src/System.Net.Sockets/tests/FunctionalTests/ReceiveMessageFrom.cs

```diff
@@ -41,5 +41,111 @@ public void Success()
                 }
             }
         }
+
+        [OuterLoop] // TODO: Issue #11345
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsSubsystemForLinux))] // https://github.com/Microsoft/BashOnWindows/issues/987
+        public void Success_IPv6()
+        {
+            if (Socket.OSSupportsIPv6)
+            {
+                using (Socket receiver = new Socket(AddressFamily.InterNetworkV6, SocketType.Dgram, ProtocolType.Udp))
+                {
+                    int port = receiver.BindToAnonymousPort(IPAddress.IPv6Loopback);
+                    receiver.SetSocketOption(SocketOptionLevel.IPv6, SocketOptionName.PacketInformation, true);
+
+                    Socket sender = new Socket(AddressFamily.InterNetworkV6, SocketType.Dgram, ProtocolType.Udp);
+                    sender.Bind(new IPEndPoint(IPAddress.IPv6Loopback, 0));
+
+                    for (int i = 0; i < TestSettings.UDPRedundancy; i++)
+                    {
+                        sender.SendTo(new byte[1024], new IPEndPoint(IPAddress.IPv6Loopback, port));
+                    }
+
+                    IPPacketInformation packetInformation;
+                    SocketFlags flags = SocketFlags.None;
+                    EndPoint remoteEP = new IPEndPoint(IPAddress.IPv6Any, 0);
+
+                    int len = receiver.ReceiveMessageFrom(new byte[1024], 0, 1024, ref flags, ref remoteEP, out packetInformation);
+
+                    Assert.Equal(1024, len);
+                    Assert.Equal(sender.LocalEndPoint, remoteEP);
+                    Assert.Equal(((IPEndPoint)sender.LocalEndPoint).Address, packetInformation.Address);
+
+                    sender.Dispose();
```

Comment:
Dispose is better.  As with other test code here, I just copied the IPv4 version and modified it for IPv6.